### ISSUE TITLE
Guidelines for handling unlabeled points

### ISSUE THREAD

**Issue by accoumar12** [filaPro/oneformer3d#96]
Hi,
I first want to thank you for your amazing model, which we already successfully applied in CAD piece segmentation to retrieve the recto, verso and deburring of sheets metal. Here is some result of the segmentation of a single file.
![image](https://github.com/user-attachments/assets/76ce154e-e3c1-462b-b17e-40b0b6858b60)
Now we want to extend to other morphologies, such as detecting the holes in the pieces. However, we do not know how to handle gracefully the unlabeled data points for such point clouds.
We first added a class called unlabeled, but the results are not satisfying... Do you recommend anything ? Be aware that most of our points are actually unlabeled so discarding them from the training would be really interesting.

Thanks for any help !

**Comment by filaPro** [comment]
are you solving instance or semantic segmentation? for semantic segmentation you simply should not calculate loss for unlabeled points. for instance segmentation unlabeled points are also not a problem, as for example on scannet we consider walls and floors unlabeled for instance segmentation

**Comment by accoumar12** [comment]
I am solving instance segmentation.
How to translate the fact that a point is unlabeled ? Giving it an arbitrary label Index ?

**Comment by filaPro** [comment]
Yes I think if your valid classes are 0, ... n_classes - 1, than given a [n_classes](https://github.com/filaPro/oneformer3d/blob/main/oneformer3d/transforms_3d.py#L123) label for a superpoint will result in ignoring it in instance segmentation.

**Comment by accoumar12** [comment]
I am following the s3dis template configuration, and i get errors when using a smaller num_instance_classes than the num_semantic_classes.

**Comment by accoumar12** [comment]
For anyone interested, I found a solution to my problem. When using the S3Dis configuration, the PointInstClassMapping_ only takes as argument the num_instance_classes. So also -1 values are assigned to the max label, which makes torch one_hot function not working as expected !

**Comment by ashutoshsingh0223** [comment]
So what was the solution? @accoumar12
